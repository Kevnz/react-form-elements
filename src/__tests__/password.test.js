import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'

import Password from '../password'

describe('Password component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const labelText = 'My Password Label'
      const { getByText, getByLabelText, container } = render(
        <Password label={labelText} />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText(labelText)
      const input = getByLabelText(labelText)
      expect(label.textContent).toBe(labelText)
      expect(input.type).toBe('password')
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in but not be valid', () => {
      const { getByLabelText, container } = render(
        <Password label="My Example Label" initialValue="password" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Example Label')
      expect(input.value).toBe('password')
      expect(input.validity.valid).toBe(true)
      fireEvent.change(input, { target: { value: 'Updated Value' } })
      expect(input.validity.valid).toBe(true)
    })
  })
})
