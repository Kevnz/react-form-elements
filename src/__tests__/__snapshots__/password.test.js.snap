// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My Password Label
  </label>
  <input
    id="id-1-1"
    name="TextBox"
    type="password"
    value=""
  />
</div>
`;

exports[`Password component handling input should set a value when passed in but not be valid 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Example Label
  </label>
  <input
    id="id-2-1"
    name="TextBox"
    type="password"
    value="password"
  />
</div>
`;
